docker/runtime/bash/http-trigger/kubeless-shell.py

```diff
@@ -1,16 +1,22 @@
 #!/usr/bin/env python
 
-from subprocess import call
+from subprocess import Popen, PIPE, STDOUT
 import bottle
 import subprocess
 import os
 import prometheus_client as prom
 
 app = application = bottle.app()
 
+#def func(osparam):
+#    cmdline = 'source /kubeless/%s.sh && %s %s' % (os.getenv('MOD_NAME'), os.getenv('FUNC_HANDLER'), osparam if osparam else '')
+#    subprocess.check_call('bash -c "' + cmdline + '"', shell=True)
+
 def func(osparam):
-    cmdline = 'source /kubeless/%s.sh && %s %s' % (os.getenv('MOD_NAME'), os.getenv('FUNC_HANDLER'), osparam if osparam else '')
-    subprocess.check_call('bash -c "' + cmdline + '"', shell=True)
+    cmdline = '/bin/bash -c "source /kubeless/%s.sh && %s %s "' % (os.getenv('MOD_NAME'), os.getenv('FUNC_HANDLER'), osparam if osparam else '')
+    proc = Popen(cmdline, stdout=PIPE, stderr=STDOUT, shell=True)
+    out = proc.communicate()[0]
+    return out.decode('utf-8')
```

Comment:
Huh, why are we forcing this into a utf-8 string?  How can we not do that?

I'm imagining that shell-based Functions will be useful for lots of things that are trivial wrappers around complex shell commands, and will typically be doing non-string things like compression, media format conversions, etc.  Those things are almost never text, and I was hoping we could just return arbitrary binary data here (and ditto for the POST request body).